src/docfx/build/markdown/BuildMarkdown.cs

```diff
@@ -37,15 +40,40 @@ internal static class BuildMarkdown
                 WordCount = wordCount,
                 Locale = locale,
                 TocRelativePath = tocMap.FindTocRelativePath(file),
-                Id = file.Id.docId,
-                VersionIndependentId = file.Id.versionIndependentId,
+                Id = id,
+                VersionIndependentId = versionIndependentId,
             };
 
             // TODO: make build pure by not output using `context.Report/Write/Copy` here
-            context.Report(file, markup.Errors);
+            var errors = error == null ? markup.Errors : markup.Errors.Concat(new[] { error });
+            context.Report(file, errors);
             context.WriteJson(model, file.OutputPath);
 
             return Task.FromResult(dependencyMapBuilder.Build());
+
+            (string id, string versionIndependentId, DocfxException error) GetIds()
```

Comment:
put this to `RedirectionMap` to share it with `BuildSchemaDocument`

Reply:
Convention from golang, put error as the first argument, to make errors significant